### PR TITLE
create test for listener and fix service mapping

### PR DESCRIPTION
**Pull request by zgorzalyj**

### CODE REVIEW

File: src/main/java/com/example/finance/controller/BudgetController.java [hunk 1]
```diff
@@ -38,8 +38,8 @@ public ResponseEntity<BudgetDto> create(@RequestBody BudgetDto budgetDto) {
     }
 
     @PutMapping(value = ApiEndpoints.Endpoints.ID)
-    public ResponseEntity<BudgetDto> update(@PathVariable UUID id, @RequestBody BudgetDto budget) {
-        BudgetDto budgetDto = budgetService.updateBudget(id, budget);
+    public ResponseEntity<BudgetDto> update(@RequestBody BudgetDto budget) {
+        BudgetDto budgetDto = budgetService.updateBudget(budget);
         return ResponseEntity.ok(budgetDto);
```
Comment (adamos98) [inline src/main/java/com/example/finance/controller/BudgetController.java:43]: You can put it in one line, because you does not do anything with `budgetDto` :)

File: src/main/java/com/example/finance/service/BudgetService.java [hunk 2]
```diff
@@ -69,12 +69,11 @@ public void saveAll(List<BudgetEntity> list) {
...
-    public BudgetDto updateBudget(UUID id, BudgetDto budgetDto) {
-        BudgetEntity budgetDb = budgetRepository.findById(id)
-                .orElseThrow(() -> new BackendException(BUDGET_NOT_FOUND_EXCEPTION_MESSAGE + id));
+    public BudgetDto updateBudget(BudgetDto budgetDto) {
+        BudgetEntity budgetDb = budgetMapper.toEntity(budgetDto);
         budgetDb.setAmount(budgetDto.amount());
-        budgetDb.setMonth(budgetDb.getMonth());
-        budgetDb.setYear(budgetDb.getYear());
+        budgetDb.setMonth(budgetDto.month());
+        budgetDb.setYear(budgetDto.year());
```
Comment (adamos98) [inline src/main/java/com/example/finance/service/BudgetService.java:76]: Those fields should be set by a mapper :)

File: src/main/java/com/example/finance/listener/BudgetEntityListener.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package com.example.finance.listener;
+
+import com.example.finance.model.entity.BudgetEntity;
+import jakarta.persistence.*;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BudgetEntityListener {
```
Comment (adamos98) [inline src/main/java/com/example/finance/listener/BudgetEntityListener.java:9]: That listener was developed only for our meeting purpose. If you don't have any reason to keep it, delete it :)

File: src/main/java/com/example/finance/service/CategoryService.java [hunk 4]
```diff
@@ -64,8 +64,7 @@ public CategoryDto create(CategoryDto category) {
 
     @Transactional
     public CategoryDto updateCategory(UUID id, CategoryDto category) {
-        CategoryEntity categoryDb = categoriesRepository.findById(id)
-                .orElseThrow(() -> new BackendException(MESSAGE));
+        CategoryEntity categoryDb = categoryMapper.toEntity(category);
         categoryDb.setName(category.name());
         categoryDb.setTransactionType(category.transactionType());
         categoryDb.setDescription(category.description());
```
Comment (adamos98) [inline src/main/java/com/example/finance/service/CategoryService.java:70]: those fields should be set by a mapper

File: src/main/java/com/example/finance/service/TransactionService.java [hunk 5]
```diff
@@ -59,8 +59,7 @@ public TransactionDto create(TransactionDto transaction) {
 
     @Transactional
     public TransactionDto updateTransaction(UUID id, TransactionDto transaction) {
-        TransactionsEntity transactionDb = transactionsRepository.findById(id)
-                .orElseThrow(() -> new BackendException(MESSAGE));
+        TransactionsEntity transactionDb = transactionMapper.toEntity(transaction);
         transactionDb.setAmount(transaction.amount());
         transactionDb.setTransactionType(transaction.transactionType());
         transactionDb.setDescription(transaction.description());
```
Comment (adamos98) [inline src/main/java/com/example/finance/service/TransactionService.java:65]: those fields should be set by a mapper

File: src/main/java/com/example/finance/service/UserAccountService.java [hunk 6]
```diff
@@ -77,7 +78,8 @@ public UserAccountDto deleteUser(UUID id) {
     }
 
     private void updateLastLogin(UserAccountEntity userAccountEntity) {
-        userAccountEntity.setLastLogin(LocalDateTime.now());
+        userAccountEntity.setEmail("test@test");
+        userAccountEntity.setLoginAction(true);
```
Comment (adamos98) [inline src/main/java/com/example/finance/service/UserAccountService.java:82]: those was only for meeting puropse

File: src/test/java/com/example/finance/service/BudgetServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
+package com.example.finance.service;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BudgetServiceTest {
```
Comment (adamos98) [inline src/test/java/com/example/finance/service/BudgetServiceTest.java:7]: If you adding a test class, please do not commit it with empty methods

File: src/test/java/com/example/finance/listener/BudgetEntityListenerTest.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+
+import static org.mockito.Mockito.verify;
+
+@SpringBootTest
+@ContextConfiguration(classes = {BudgetEntityListener.class})
+class BudgetEntityListenerTest {
```
Comment (adamos98) [inline src/test/java/com/example/finance/listener/BudgetEntityListenerTest.java:16]: I think you don't need to test that class because it was only for our meeting purpose. If you don't have any idea to create your own listener - let me know. I will figure out something :)

**Review by adamos98 — CHANGES_REQUESTED** [review]
